cupy/cuda/cupy_thrust.h

```diff
@@ -8,63 +8,29 @@ namespace cupy {
 
 namespace thrust {
 
-template <typename T>
-void _sort(void *, size_t *, const std::vector<ptrdiff_t>&, intptr_t, void *);
-
-template <typename T>
-void _lexsort(size_t *, void *, size_t, size_t, intptr_t, void *);
-
-template <typename T>
-void _argsort(size_t *, void *, void *, const std::vector<ptrdiff_t>&, intptr_t,
-              void *);
-
-/*
-   The functions with the suffix _fp16 are used only when certain conditions are met
-*/
-void _sort_fp16(void *, size_t *, const std::vector<ptrdiff_t>&, intptr_t, void *);
-
-void _lexsort_fp16(size_t *, void *, size_t, size_t, intptr_t, void *);
-
-void _argsort_fp16(size_t *, void *, void *, const std::vector<ptrdiff_t>&, intptr_t,
-                   void *);
+void thrust_sort(int, void *, size_t *, const std::vector<ptrdiff_t>&, intptr_t, void *);
+void thrust_lexsort(int, size_t *, void *, size_t, size_t, intptr_t, void *);
+void thrust_argsort(int, size_t *, void *, void *, const std::vector<ptrdiff_t>&, intptr_t, void *);
 
 } // namespace thrust
 
 } // namespace cupy
 
 #else // CUPY_NO_CUDA
 
-#include "../../cupy_backends/cuda/cupy_common.h"
 #define THRUST_VERSION 0
 
 namespace cupy {
 
 namespace thrust {
```

Comment:
Now I think these namespaces do not make much sense. How about removing them like `cupy_cub.h`?

Reply:
Yeah sounds good. Do you remember why we needed the namespaces?

Reply:
I initially introduced the namespace but was not necessary. It makes no harm to remove it.